test/icc-x-api/icc-contact-x-api.ts

```diff
@@ -189,4 +189,41 @@ describe('icc-x-contact-api Tests', () => {
     assert(foundServices.rows![0].id == createdContact.services![0].id)
     assert(foundServices.rows![0].healthElementsIds!.find((heId) => heId == healthElement!.id!) != undefined)
   })
+
+  it('contacts findBy Using Post for HCP', async () => {
+    // Given
+    const {
+      userApi: userApiForHcp,
+      dataOwnerApi: dataOwnerApiForHcp,
+      patientApi: patientApiForHcp,
+      contactApi: contactApiForHcp,
+      healthcareElementApi: hElementApiForHcp,
+      cryptoApi: cryptoApiForHcp,
+    } = await Api(env!.iCureUrl, env!.dataOwnerDetails[hcp1Username].user, env!.dataOwnerDetails[hcp1Username].password, crypto)
+
+    const hcpUser = await userApiForHcp.getCurrentUser()
+    await initKey(dataOwnerApiForHcp, cryptoApiForHcp, hcpUser, env!.dataOwnerDetails[hcp1Username].privateKey)
+
+    const patient = (await createPatient(patientApiForHcp, hcpUser)) as Patient
+    const healthElement = await createHealthElement(hElementApiForHcp, hcpUser, patient)
+    const contactToCreate = await createBasicContact(contactApiForHcp, hcpUser, patient).then((contact) => {
```

Comment:
I don't really get how this test can work: `createBasicContact` is just calling the newInstance() method of the API. To create a contact in the backend, you still need to call `createContactWithUser`.